### PR TITLE
Add glib::Priority as custom type, replace priorities in async functi…

### PR DESCRIPTION
**Pull request by EPashkin**
…ons with it

Part of https://github.com/gtk-rs/gio/pull/85

For gio need add `"GLib.Priority",` in `manual` array in `Gir.toml`

cc @GuillaumeGomez, @sdroege, @MathieuDuponchelle

### CODE REVIEW

**Comment by EPashkin** [comment]
Dirty hack.
Currently has effect only on gio

**Comment by EPashkin** [comment]
Diff
```diff
@@ -742,7 +741,7 @@ impl<O: IsA<File>> FileExt for O {
         }
     }
 
-    fn read_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<FileInputStream, Error>) + Send + 'static>(&self, io_priority: i32, cancellable: P, callback: Q) {
+    fn read_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<FileInputStream, Error>) + Send + 'static>(&self, io_priority: glib::Priority, cancellable: P, callback: Q) {
         let cancellable = cancellable.into();
         let cancellable = cancellable.to_glib_none();
         let user_data: Box<Box<Q>> = Box::new(Box::new(callback));
@@ -756,7 +755,7 @@ impl<O: IsA<File>> FileExt for O {
         }
         let callback = read_async_trampoline::<Q>;
         unsafe {
-            ffi::g_file_read_async(self.to_glib_none().0, io_priority, cancellable.0, Some(callback), Box::into_raw(user_data) as *mut _);
+            ffi::g_file_read_async(self.to_glib_none().0, io_priority.to_glib(), cancellable.0, Some(callback), Box::into_raw(user_data) as *mut _);
         }
     }
 ```

**Comment by sdroege** [comment]
Ugly but probably worth it. Results look good to me, chances for this being applied to something else seem very low :)

File: src/custom_type_glib_priority.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+struct ReplaceToPriority {
+    pub tid_priority: TypeId,
+    pub tid_int: TypeId,
+}
+
+impl FunctionsMutVisitor for ReplaceToPriority {
+    fn visit_function_mut(&mut self, func: &mut Function) -> bool {
```
Comment (GuillaumeGomez) [inline src/custom_type_glib_priority.rs]: This function always returns `true` XD.

Comment (EPashkin) [reply src/custom_type_glib_priority.rs]: Currently yes, but IMHO possibility to stop visiting is will need sometime.

Comment (EPashkin) [reply src/custom_type_glib_priority.rs]: Can remove return type if you prefer without it

**Comment by EPashkin** [comment]
Also changes in `gtk_source_file_loader_load_async` but it commented by different reasons

File: src/custom_type_glib_priority.rs (hunk 1 above)
Comment (EPashkin) [reply src/custom_type_glib_priority.rs]: @GuillaumeGomez Or I merge PR as is ?

Comment (GuillaumeGomez) [reply src/custom_type_glib_priority.rs]: I don't mind. Maybe add a `TODO` about it so it's not forgotten?

Comment (EPashkin) [reply src/custom_type_glib_priority.rs]: Added

**Pull request merged**